File: recordOfProblems/MergeTwoSortedLists.java
public class MergeTwoSortedLists {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode result = new ListNode();
        ListNode resultHead = result;
        ListNode current1 = list1;
        ListNode current2 = list2;
        
        while(current1 != null || current2 != null) {
            int val1 = 200;
            int val2 = 200;
            
            if(current1 != null) {
                val1 = current1.val;
            }
            if(current2 != null) {
                val2 = current2.val;
            }
            
            if(val1 < val2) {
                result.next = new ListNode(val1);
                current1 = current1.next;
            }
            else {
                result.next = new ListNode(val2);
                current2 = current2.next;
            }
            result = result.next;
        }
        
        return resultHead.next;
    }
}

File: recordOfProblems/c++/MergeTwoSortedLists.cc
ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
    ListNode* sol = new ListNode(0);
    ListNode* curr = sol;
    
    while(list1 != nullptr || list2 != nullptr) {
        ListNode* next;
        if(list1 == nullptr) {
            next = new ListNode(list2->val);
            list2 = list2->next;
        }
        else {
            if(list2 == nullptr) {
                next = new ListNode(list1->val);
                list1 = list1->next;
            }
            else {
                if(list1->val < list2->val) {
                    next = new ListNode(list1->val);
                    list1 = list1->next;
                }
                else {
                    next = new ListNode(list2->val);
                    list2 = list2->next;
                }
            }
        }
        curr->next = next;
        curr = curr->next;
    }
    
    return sol->next;
}
